src/shared/regions/defaultRegionProvider.ts

```diff
@@ -4,98 +4,59 @@
  */
 
 import * as vscode from 'vscode'
-import { endpointsFileUrl } from '../constants'
-import { ext } from '../extensionGlobals'
-import { getLogger, Logger } from '../logger'
-import { CompositeResourceFetcher } from '../resourcefetcher/compositeResourceFetcher'
-import { FileResourceFetcher } from '../resourcefetcher/fileResourceFetcher'
-import { HttpResourceFetcher } from '../resourcefetcher/httpResourceFetcher'
-import { ResourceFetcher } from '../resourcefetcher/resourcefetcher'
+import { Endpoints, Region } from './endpoints'
+import { EndpointsProvider } from './endpointsProvider'
 import { RegionInfo } from './regionInfo'
 import { RegionProvider } from './regionProvider'
 
-export interface RawRegion {
-    description: string
-}
+export class DefaultRegionProvider implements RegionProvider {
+    private readonly onRegionProviderUpdatedEmitter: vscode.EventEmitter<void> = new vscode.EventEmitter()
+    private _loadedRegions: RegionInfo[] = []
 
-export interface RawPartition {
-    partition: string
-    regions: {
-        [regionKey: string]: RawRegion
+    public constructor(endpointsProvider: EndpointsProvider) {
+        endpointsProvider.onEndpointsUpdated(e => this.loadFromEndpointsProvider(e))
+        this.loadFromEndpointsProvider(endpointsProvider)
     }
-}
-
-export interface RawEndpoints {
-    partitions: RawPartition[]
-}
 
-export class DefaultRegionProvider implements RegionProvider {
-    private readonly logger: Logger = getLogger()
-    private _areRegionsLoaded: boolean = false
-    private _loadedRegions: RegionInfo[]
-    private readonly _resourceFetcher: ResourceFetcher
-
-    public constructor(resourceFetcher: ResourceFetcher) {
-        this._loadedRegions = []
-        this._resourceFetcher = resourceFetcher
+    public get onRegionProviderUpdated(): vscode.Event<void> {
+        return this.onRegionProviderUpdatedEmitter.event
     }
 
-    // Returns an array of Regions, and caches them in memory.
     public async getRegionData(): Promise<RegionInfo[]> {
-        if (this._areRegionsLoaded) {
-            return this._loadedRegions
-        }
-
-        let availableRegions: RegionInfo[] = []
-        try {
-            this.logger.info('Retrieving AWS endpoint data')
-
-            const endpointsContents = await this._resourceFetcher.get()
-
-            if (!endpointsContents) {
-                throw new Error('No endpoints data found')
-            }
-
-            const allEndpoints = JSON.parse(endpointsContents) as RawEndpoints
+        return this._loadedRegions
```

Comment:
curious why this returns a promise? Would we ever want to refresh or do something complicated here?

Reply:
That's a holdover from previous code. In my next PR I plan to remove this method and add new methods that surface partition-region-service relationships. Rather than ripple updates to all the calls to remove async in this change, I decided to leave it as is.